internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1004,6 +1004,48 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				}, false),
 			},
 
+			"storage_blob_driver": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  false,
```

Comment:
Can we rename this to (also after it's been nested under a `storage_profile` block)
```suggestion
			"blob_driver_enabled": {
				Type:     pluginsdk.TypeBool,
				Optional: true,
				Default:  false,
```